usaspending_api/database_scripts/etl/transaction_delta_view.sql

```diff
@@ -116,6 +124,14 @@ LEFT JOIN subtier_agency SAA ON (AA.subtier_agency_id = SAA.subtier_agency_id)
 LEFT JOIN toptier_agency TFA ON (FA.toptier_agency_id = TFA.toptier_agency_id)
 LEFT JOIN subtier_agency SFA ON (FA.subtier_agency_id = SFA.subtier_agency_id)
 LEFT JOIN references_cfda CFDA ON (FABS.cfda_number = CFDA.program_number)
+LEFT JOIN (
```

Comment:
Concerning recipient "hashes" + level baked into the view and index:

1. Currently the matviews do not store the level (-P, -R, -C) on the hash. I don't recall why. Perhaps it helps with aggregations.
2. This logic doesn't handle the "special case" recipients. These recipients are excluded from the Recipient Profile because their purpose is to obscure the recipients for privacy (or similar) reasons. In existing views the recipient links are often created in the business logic after data are fetched from the DB. See this subquery expression as an example for handling edge-cases: https://github.com/fedspendingtransparency/usaspending-api/blob/dev/usaspending_api/common/recipient_lookups.py#L138-L149 and this example of combining the full recipient  hash + level after the records are returned by the DB: https://github.com/fedspendingtransparency/usaspending-api/blob/dev/usaspending_api/search/v2/views/spending_by_category.py#L283


I don't know the best design for recipient links, someone with more time can dig into the data and make cleaner logic. Until then, it might be better to follow a similar pattern

Reply:
Made some changes to the `recipient_id` filter and the view SQL that make it function the same way only with ES instead of Django. 

Regarding the special cases, from my understanding, it should allow for the same functionality as the current `spending_by_category` implementation now.